src/screen/ConnectionsList/NewConnectionModal/index.tsx

```diff
@@ -62,12 +68,16 @@ const NewConnectionModal: React.FC<SharedModalProps> = ({
         const { name, host, port, password } = data
 
         try {
-          const connections = saveAndGetConnections({
+          const newConnectionData = {
             name,
             host,
             port: Number(port),
             password
-          })
+          }
+
+          const connections = connectionToEdit
+            ? updateAndGetConnections(connectionToEdit, newConnectionData)
+            : saveAndGetConnections(newConnectionData)
```

Comment:
Maybe we should change the `saveAndGetConnections` method name to `createAndGetConnections`

Reply:
Thought about that too. :smile: